src/Cake.Common/Build/TeamCity/TeamCityDisposableHelpers.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Cake.Common.Build.TeamCity
+{
+    /// <summary>
+    /// Disposable helper for writing TeamCity message blocks.
+    /// </summary>
+    public sealed class TeamCityBlock : IDisposable
```

Comment:
1. I would actually want to avoid adding functionality for the purpose of being reused. I would prefer that we implement stuff without usability in mind, and do this as a refactoring part later on i.e. if we see that a lot of stuff sprinkled across the project/solution uses the same mecanism.

2. I think we should choose whether or not to use the disposable pattern or Start/End. If we want to give the option to the user to choose, I would rather see Start/End implemented on the provider, and add specialized extensionmethods on top of it that extends the ITeamCityProvider with the IDisposable stuff (using the Start/End methods).